lib/kinesis_publisher.rb

```diff
@@ -1,23 +1,25 @@
 class KinesisPublisher
-  def self.publish(event_type, partition_key, data)
+  def self.publish(event_type, partition_key, data, linked = {})
     return unless client
 
     client.put_record(
       stream_name: stream_name,
       partition_key: partition_key.to_s,
-      data: format_data(event_type, data)
+      data: format_data(event_type, data, linked)
     )
   rescue StandardError => ex
     # While we're still evaluating Kinesis, don't propagate this error, but do notify Honeybadger.
     Honeybadger.notify(ex)
   end
 
-  def self.format_data(event_type, data)
+  def self.format_data(event_type, data, linked)
     {
       source: 'panoptes',
       type: event_type,
       version: '1.0.0',
-      data: data
+      timestamp: Time.now.utc.iso8601,
```

Comment:
I assume we're tracking times here and we use the payload timestamp for actual event tracking?